Fw/Logger/Logger.hpp

```diff
@@ -4,87 +4,59 @@
-             * \param a0: zeroth argument. (Default: 0)
-             * \param a1: first argument. (Default: 0)
-             * \param a2: second argument. (Default: 0)
-             * \param a3: third argument. (Default: 0)
-             * \param a4: fourth argument. (Default: 0)
-             * \param a5: fifth argument. (Default: 0)
-             * \param a6: sixth argument. (Default: 0)
-             * \param a7: seventh argument. (Default: 0)
-             * \param a8: eighth argument. (Default: 0)
-             * \param a9: ninth argument. (Default: 0)
-             */
-            virtual void log(
-                const char* fmt,
-                POINTER_CAST a0 = 0,
-                POINTER_CAST a1 = 0,
-                POINTER_CAST a2 = 0,
-                POINTER_CAST a3 = 0,
-                POINTER_CAST a4 = 0,
-                POINTER_CAST a5 = 0,
-                POINTER_CAST a6 = 0,
-                POINTER_CAST a7 = 0,
-                POINTER_CAST a8 = 0,
-                POINTER_CAST a9 = 0
-            ) = 0;
+class Logger {
+    friend struct LoggerRules::Register;
 
-            /**
-             * Logs a message using the currently specified static logger. If a logger is not
-             * registered, then the log message is dropped.
-             * \param fmt: format string in which to place arguments
-             * \param a0: zeroth argument. (Default: 0)
-             * \param a1: first argument. (Default: 0)
-             * \param a2: second argument. (Default: 0)
-             * \param a3: third argument. (Default: 0)
-             * \param a4: fourth argument. (Default: 0)
-             * \param a5: fifth argument. (Default: 0)
-             * \param a6: sixth argument. (Default: 0)
-             * \param a7: seventh argument. (Default: 0)
-             * \param a8: eighth argument. (Default: 0)
-             * \param a9: ninth argument. (Default: 0)
-             */
-            static void logMsg(
-                const char* fmt,
-                POINTER_CAST a0 = 0,
-                POINTER_CAST a1 = 0,
-                POINTER_CAST a2 = 0,
-                POINTER_CAST a3 = 0,
-                POINTER_CAST a4 = 0,
-                POINTER_CAST a5 = 0,
-                POINTER_CAST a6 = 0,
-                POINTER_CAST a7 = 0,
-                POINTER_CAST a8 = 0,
-                POINTER_CAST a9 = 0
-            );
+  public:
+    //! \brief log a formated string with supplied arguments
+    //!
+    //! Logs a format string with the arguments filled-in. This delegates to StringBase.format, which in-turn
+    //! delegates to snprintf. This implies that the caller is fully responsible for handling the type safety of
+    //! the supplied format string. The format string uses C-style (printf function family) formatting.
+    //! \param format: format string
+    //! \param ...: var-args list of arguments to inject into format string.
+    static void log(const char* format, ...);
```

Comment:
Could we be interested in aliasing this to `logMsg(format, ...)` as well for backwards compatibility? Otherwise we should flag this PR as a breaking change for release notes

Reply:
nevermind ... just saw it has the label already

Reply:
The old format was very dangerous, so we want to force users to update with consideration.  Typically, this is a better approach except when the old way is dangerous.